pkg/reconciler/pipelinerun/resources/pipelinerunstate.go

```diff
@@ -32,6 +32,12 @@ import (
 // state of the PipelineRun.
 type PipelineRunState []*ResolvedPipelineRunTask
 
+type PipelineRunFacts struct {
+	State           PipelineRunState
+	TasksGraph      *dag.Graph
+	FinalTasksGraph *dag.Graph
+}
+
```

Comment:
Given that `PipelineRunState` is a list of `ResolvedPipelineRunTask`... https://github.com/tektoncd/pipeline/blob/f412b1189b20721c2236b5017fc20f9458ac9b7d/pkg/reconciler/pipelinerun/resources/pipelinerunstate.go#L33

...I was wondering if we could make that explicit by calling it `ResolvedPipelineRunTasks` then we can call this object `PipelineRunState` instead of  `PipelineRunFacts`, what do you think?

Reply:
@jerop I like it but given `ResolvedPipelineRunTask` is so long and differentiating it with the slice of `ResolvedPipelineRunTask` by just additional `s` might confuse readers.

Reply:
we could rename `ResolvedPipelineRunTask` to `ResolvedPipelineTask`? 😜

Reply:
thanks @jerop, will have that renaming going in a separate PR. Once this is merged, would like to get the [refactoring](https://github.com/pritidesai/pipeline/commit/61b9e7ed24a8272d0f84da5c9a33bcc26bbdcb80) of `GetPipelineConditionStatus` in 🙏